drivers/dma/dma_mcux_lpc.c

```diff
@@ -314,12 +324,31 @@ static int dma_mcux_lpc_configure(const struct device *dev, uint32_t channel,
 		}
 	} else {
 		/* block_count shall be 1 */
-		/* Only one buffer, enable interrupt */
-		xferConfig = DMA_CHANNEL_XFER(0UL, 0UL, 1UL, 0UL,
-					config->dest_data_size,
-					src_inc,
-					dst_inc,
-					block_config->block_size);
+		/* If DMA Channel Chaining is expected, then the
+		 * burst length must be setup differently aswell.
+		 */
+		if ((config->dest_chaining_en && config->source_chaining_en) ||
+		   (!config->dest_chaining_en && !config->source_chaining_en)) {
```

Comment:
can we add a comment block in each of the condition blocks to describe what is going on?

Reply:
Thank you, I have added some additional comments, please let me know if this is more helpful.